### PR TITLE
Roles / Claims Transformations: only add the roles once

### PR DESCRIPTION
**Pull request by JimHume**
This avoids having an infinite number of roles get added to the ClaimsPrincipal when mapping the resource/realm access to roles.

This is in reference to https://github.com/NikiforovAll/keycloak-authorization-services-dotnet/issues/33

### CODE REVIEW

**Review by rasmus234 — APPROVED** [review]

File: tests/Keycloak.AuthServices.Authentication.Tests/Usings.cs [hunk 1]
```diff
@@ -0,0 +1 @@
+global using Microsoft.VisualStudio.TestTools.UnitTesting;
```
Comment (NikiforovAll) [inline tests/Keycloak.AuthServices.Authentication.Tests/Usings.cs:1]: What is the purpose of adding this global using?

File: tests/Keycloak.AuthServices.Authentication.Tests/Claims/KeycloakRolesClaimsTransformationTests.cs [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+    public async Task ClaimsTransformationShouldMap(RolesClaimTransformationSource roleSource)
+    {
+        // create the service
+        var target = new KeycloakRolesClaimsTransformation(ClaimTypes.Role, roleSource, MyAudience);
+
+        // The fixture
+        var claimsPrincipal = GetClaimsPrincipal();
+
+        // This should work many times
+        foreach (var _ in Enumerable.Range(1, 3))
```
Comment (NikiforovAll) [inline tests/Keycloak.AuthServices.Authentication.Tests/Claims/KeycloakRolesClaimsTransformationTests.cs:22]: Let's change it to "for" loop style.

File: tests/Keycloak.AuthServices.Authentication.Tests/Claims/KeycloakRolesClaimsTransformationTests.cs [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+            claimsPrincipal = await target.TransformAsync(claimsPrincipal);
+            claimsPrincipal.HasClaim(ClaimTypes.Role, MyFirstClaim).Should().BeTrue();
+            claimsPrincipal.HasClaim(ClaimTypes.Role, MySecondClaim).Should().BeTrue();
+            claimsPrincipal.Claims.Count(item => ClaimTypes.Role == item.Type).Should().Be(2);
+        }
+    }
+
+    // The resource_access claim type
+    private const string MyResourceClaimType = "resource_access";
+    private const string MyResourceClaimValue = $"{{\u0022{MyAudience}\u0022:{{\u0022roles\u0022:[\u0022{MyFirstClaim}\u0022,\u0022{MySecondClaim}\u0022]}}}}";
```
Comment (NikiforovAll) [inline tests/Keycloak.AuthServices.Authentication.Tests/Claims/KeycloakRolesClaimsTransformationTests.cs:33]: Let's use regular double quote escaping to improve readability. $"\"x\"" or @"""x"""

File: tests/Keycloak.AuthServices.Authentication.Tests/Keycloak.AuthServices.Authentication.Tests.csproj [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+    <TargetFramework>net6.0</TargetFramework>
+    <ImplicitUsings>enable</ImplicitUsings>
+    <Nullable>enable</Nullable>
+
+    <IsPackable>false</IsPackable>
+  </PropertyGroup>
+
+  <ItemGroup>
+    <PackageReference Include="Microsoft.NET.Test.Sdk" Version="17.1.0" />
+    <PackageReference Include="MSTest.TestAdapter" Version="2.2.8" />
```
Comment (NikiforovAll) [inline tests/Keycloak.AuthServices.Authentication.Tests/Keycloak.AuthServices.Authentication.Tests.csproj:13]: We don't really need this one. We are using XUnit as a default testing framework for this project.

**Review by NikiforovAll — CHANGES_REQUESTED** [review]
Thank you for the MR. I'm looking forward to working on it with you.

**Comment by JimHume** [comment]
All requested changes have been pushed.

**Comment by NikiforovAll** [comment]
@JimHume  thank you for the contribution. I appreciate it 🙏

**Pull request merged**